### DISCUSSION TITLE
Using alembic after upgrading to postgresql 16 results to error

### DISCUSSION THREAD

**Discussion by Kotsakis** [sqlalchemy/alembic#1579] [Usage Questions] [answered]
Hello,

I am getting this error when I execute `alembic -c capacity/infrastructure/alembic.ini revision --autogenerate -m 'remove_unecessary_relationships_part_1'`

```
File "/Users/ekotsal/Library/Caches/pypoetry/virtualenvs/api-zHdMukpK-py3.12/lib/python3.12/site-packages/sqlalchemy/dialects/postgresql/psycopg2.py", line 690, in import_dbapi
    import psycopg2
  File "/Users/ekotsal/Library/Caches/pypoetry/virtualenvs/api-zHdMukpK-py3.12/lib/python3.12/site-packages/psycopg2/__init__.py", line 51, in <module>
    from psycopg2._psycopg import (                     # noqa
ImportError: dlopen(/Users/ekotsal/Library/Caches/pypoetry/virtualenvs/api-zHdMukpK-py3.12/lib/python3.12/site-packages/psycopg2/_psycopg.cpython-312-darwin.so, 0x0002): Library not loaded: /opt/homebrew/opt/postgresql@15/lib/libpq.5.dylib
  Referenced from: <546BCD40-B0AA-30AC-AA38-5E88389FDC9E> /Users/ekotsal/Library/Caches/pypoetry/virtualenvs/api-zHdMukpK-py3.12/lib/python3.12/site-packages/psycopg2/_psycopg.cpython-312-darwin.so
  Reason: tried: '/opt/homebrew/Cellar/postgresql/16.6/lib/libpq.5.dylib' (no such file), '/libpq.5.dylib' (no such file), '/opt/homebrew/opt/postgresql@15/lib/libpq.5.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/opt/homebrew/opt/postgresql@15/lib/libpq.5.dylib' (no such file), '/opt/homebrew/opt/postgresql@15/lib/libpq.5.dylib' (no such file)
```

As i wrote in the heading, I was using alembic with postgresql 15 but the problem started after upgrading to version 16. 

I installed `postgresql@16` and `libpq` using `brew`

I would appreciate any ideas to resolve this. 

Update

The file indeed exists but alembic is trying to find it in a different location, spot the `@16` missing 

![image](https://github.com/user-attachments/assets/698abbf0-8135-401f-9dc0-1cebce18de6a)

**Comment by Kotsakis** [comment, marked as answer]
Alembic is working again by setting this environment variable in my `.zshrc`. I found this information after little googling and chatgpting 

`export DYLD_LIBRARY_PATH=/opt/homebrew/Cellar/postgresql@16/16.6/lib:$DYLD_LIBRARY_PATH`

However, i still do not understand what is the root cause of this, i.e what causes alembic to look at this specific set of paths and why it misses the `@16`.

**Comment by zzzeek** [comment]
this is a driver installation issue and you'd want to consult with psycopg2 at https://github.com/psycopg/psycopg2
